src/browser/modules/D3Visualization/lib/visualization/components/mouseEventHandlers.ts

```diff
@@ -0,0 +1,146 @@
+import { D3DragEvent, drag as d3Drag } from 'd3-drag'
+import { Simulation } from 'd3-force'
+import { BaseType, Selection } from 'd3-selection'
+import { ZoomBehavior } from 'd3-zoom'
+
+import {
+  DEFAULT_ALPHA,
+  DEFAULT_ALPHA_TARGET,
+  DRAGGING_ALPHA_TARGET
+} from '../constants'
+import Relationship from './Relationship'
+import VizNode from './VizNode'
+
+export const zoomEventHandler = (
+  selection: Selection<SVGGElement, unknown, BaseType, unknown>,
+  baseGroup: Selection<SVGGElement, unknown, BaseType, unknown>,
+  zoomBehavior: ZoomBehavior<SVGGElement, unknown>
+) => {
+  const handleZoomOnShiftScroll = (e: WheelEvent) => {
+    if (e.shiftKey) {
+      e.preventDefault()
+
+      const delta =
+        -e.deltaY * (e.deltaMode === 1 ? 0.05 : e.deltaMode ? 1 : 0.002)
+
+      return zoomBehavior.scaleBy(baseGroup, 1 + delta)
+    }
+  }
+
+  return selection
+    .on('dblclick.zoom', null)
+    .on('DOMMouseScroll.zoom', handleZoomOnShiftScroll)
+    .on('wheel.zoom', handleZoomOnShiftScroll)
+    .on('mousewheel.zoom', handleZoomOnShiftScroll)
+}
+
+export const nodeEventHandlers = (
+  selection: Selection<SVGGElement, VizNode, BaseType, unknown>,
+  trigger: (event: string, node: VizNode) => void,
+  simulation: Simulation<VizNode, Relationship>
+) => {
+  let initialDragPosition: [number, number]
```

Comment:
Does it matter that these don't have an initial/default values? It seems they'd always be defined by `dragStart` before they're used in `dragged`. But I'm not sure how the handlers work